bee-crypto/src/ternary/sponge/curlp/unrolled/mod.rs

```diff
@@ -26,7 +29,10 @@ pub struct UnrolledCurlP81 {
 
 impl UnrolledCurlP81 {
     /// Creates a new [`UnrolledCurlP81`].
+    // Many safety conditions hold only if the `HASH_LENGTH` is smaller than `256`.
+    #[allow(clippy::assertions_on_constants)]
     pub fn new() -> Self {
+        assert!(HASH_LENGTH < 256);
```

Comment:
Isn't `debug_assert` enough?

Reply:
it is a hard assert because most of the safety guarantees depend on this condition. Also this is optimized away to nothing if it's true.